scripts/lib/api/Pkg.ts

```diff
@@ -167,18 +167,169 @@ export class Pkg {
+      };
+    }
+
+    // Otherwise, we have to deal with Qiskit <0.45, when we had the qiskit-metapackage comprised of
+    // multiple packages. Refer to the version table in api/qiskit/release-notes/0.44.md.
+    return (fileName) =>
+      determineHistoricalQiskitGithubUrl(
+        this.versionWithoutPatch,
+        normalizeFile(fileName),
+      );
+  }
+}
+
+const QISKIT_METAPACKAGE_TO_TERRA = new Map([
+  ["0.44", "0.25"],
+  ["0.43", "0.24"],
+  ["0.42", "0.23"],
+  ["0.41", "0.23"],
+  ["0.40", "0.23"],
+  ["0.39", "0.22"],
+  ["0.38", "0.21"],
+  ["0.37", "0.21"],
+  ["0.36", "0.20"],
+  ["0.35", "0.20"],
+  ["0.34", "0.19"],
+  ["0.33", "0.19"],
+  ["0.32", "0.18"],
+  ["0.31", "0.18"],
+  ["0.30", "0.18"],
+  ["0.29", "0.18"],
+  ["0.28", "0.18"],
+  ["0.27", "0.17"],
+  ["0.26", "0.17"],
+  ["0.25", "0.17"],
+  ["0.24", "0.16"],
+  ["0.19", "0.14"],
+]);
+const QISKIT_METAPACKAGE_TO_AER = new Map([
+  ["0.43", "0.12"],
+  ["0.42", "0.12"],
+  ["0.41", "0.11"],
+  ["0.40", "0.11"],
+  ["0.39", "0.11"],
+  ["0.38", "0.11"],
+  ["0.37", "0.10"],
+  ["0.36", "0.10"],
+  ["0.35", "0.10"],
+  ["0.34", "0.10"],
+  ["0.33", "0.9"],
+  ["0.32", "0.9"],
+  ["0.31", "0.9"],
+  ["0.30", "0.9"],
+  ["0.29", "0.8"],
+  ["0.28", "0.8"],
+  ["0.27", "0.8"],
+  ["0.26", "0.8"],
+  ["0.25", "0.8"],
+  ["0.24", "0.7"],
+  ["0.19", "0.5"],
+]);
+const QISKIT_METAPACKAGE_TO_IGNIS = new Map([
+  ["0.36", "0.7"],
+  ["0.35", "0.7"],
+  ["0.34", "0.7"],
+  ["0.33", "0.7"],
+  ["0.32", "0.6"],
+  ["0.31", "0.6"],
+  ["0.30", "0.6"],
+  ["0.29", "0.6"],
+  ["0.28", "0.6"],
+  ["0.27", "0.6"],
+  ["0.26", "0.6"],
+  ["0.25", "0.6"],
+  ["0.24", "0.5"],
+  ["0.19", "0.3"],
+]);
+const QISKIT_METAPACKAGE_TO_AQUA = new Map([
+  ["0.32", "0.9"],
+  ["0.31", "0.9"],
+  ["0.30", "0.9"],
+  ["0.29", "0.9"],
+  ["0.28", "0.9"],
+  ["0.27", "0.9"],
+  ["0.26", "0.9"],
+  ["0.25", "0.9"],
+  ["0.24", "0.8"],
+  ["0.19", "0.7"],
+]);
+const QISKIT_METAPACKAGE_TO_IBMQ_PROVIDER = new Map([
+  ["0.43", "0.20"],
+  ["0.42", "0.20"],
+  ["0.41", "0.20"],
+  ["0.40", "0.19"],
+  ["0.39", "0.19"],
+  ["0.38", "0.19"],
+  ["0.37", "0.19"],
+  ["0.36", "0.19"],
+  ["0.35", "0.18"],
+  ["0.34", "0.18"],
+  ["0.33", "0.18"],
+  ["0.32", "0.18"],
+  ["0.31", "0.17"],
+  ["0.30", "0.16"],
+  ["0.29", "0.16"],
+  ["0.28", "0.15"],
+  ["0.27", "0.14"],
+  ["0.26", "0.13"],
+  ["0.25", "0.12"],
+  ["0.24", "0.12"],
+  ["0.19", "0.7"],
+]);
+
+function determineHistoricalQiskitGithubUrl(
+  metapackageVersion: string,
+  fileName: string,
+): string {
+  const getOrThrow = (mapping: Map<string, string>): string => {
+    const result = mapping.get(metapackageVersion);
+    if (result === undefined) {
+      throw new Error(
+        `No compatible version found for the file ${fileName} with qiskit-metapackage ${metapackageVersion}}`,
+      );
+    }
+    return result;
+  };
+
+  let slug: string;
+  let version: string;
+  if (
```

Comment:
Matt T reviewed the `if` statements to make sure this covers all the import paths.